### ISSUE TITLE
Detected [E0121] in struct

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#7671]
## Environment

* **IntelliJ Rust plugin version:** 0.4.154.4041-212-nightly
* **Rust toolchain version:** 1.53.0 (53cb7b09b 2021-06-17) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.2 (CL-212.4746.93)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

error[E0121]: the type placeholder `_` is not allowed within types on item signatures for **structs**
But IDE doesn't detect it:
<img width="270" alt="изображение" src="https://user-images.githubusercontent.com/49211026/129342316-b9cd6805-ee27-4f72-8c24-a77e8a93f6f8.png">

## Steps to reproduce
```rust
struct Baz(i32, _); //error
fn foo(a: i32, _){  //same error detected
    todo!()
}
```
